### PR TITLE
Restore Python 3 compatibility

### PR DESCRIPTION
**Pull request by pschella**
Inverse PR

### CODE REVIEW

File: tests/testThreads2.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 import unittest
 
+from past.builtins import long
```
Comment (timj) [inline tests/testThreads2.py]: It's not entirely obvious to me that `getNumThreads` really needs to return a `long`.

Comment (pschella) [inline tests/testThreads2.py]: With pybind11 a C++ function with return type `int` gives a `long` on Python 2. Not sure if this is a bug or design. I have filed a ticket upstream, but in the mean time this is needed.

**Comment by timj** [comment]
Can we close this PR to avoid further confusion?

**Comment by jdswinbank** [comment]
"Further"?

Regardless, both DM-7519 & DM-6168 are done; this PR is clearly obsolete.

**Pull request closed without merging**